packages/robot_http_api/include/hardware_test_robot_host/hardware_test_robot_host.py

```diff
@@ -0,0 +1,148 @@
+        super().__init__(service_identifier="tests/wifi")
+
+        # attr
+        self._interface = wifi_interface
+
+    def test_id(self) -> str:
+        return "USB Wifi Dongle"
+
+    def test_description_preparation(self) -> str:
+        return self.html_util_ul(
+            [
+                "Make sure the USB Wifi dongle is plugged in to your Duckiebot, and it is blinking.",
+            ]
+        )
+
+    def test_description_expectation(self) -> str:
+        return self.html_util_ul(
+            [
+                "The IP address of the <code>wlan0</code> network interface should be displayed below.",
+            ]
+        )
+
+    def test_description_log_gather(self) -> str:
+        return self.html_util_ul(
+            [
+                "On your laptop, run the following command to save the logs.",
+                "Replace the <code>[path/to/save]</code> to the directory path where you would like to save the logs.",
+                "<code>ssh duckie@[ROBOT_NAME].local ifconfig > [path/to/save/]logs-network-ifconfig.txt</code>",
+                "(You might need to provide the password to your Duckiebot when prompted.)",
+            ]
+        )
+
+    def _get_ipv4_addr(self) -> str:
+        addrs = ni.ifaddresses(self._interface)
+        if ni.AF_INET in addrs:
+            return addrs[ni.AF_INET][0]["addr"]
+        return "None"
+
+    def cb_run_test(self, _):
+        rospy.loginfo(f"[{self.test_id()}] Test service called.")
+        success = True
+        response = ""
+
+        try:
+            response = self._get_ipv4_addr()
+        except Exception as e:
+            rospy.logerr(f"[{self.test_id()}] Experienced error: {e}")
+            success = False
+
+        return self.format_response_object(
+            success=success,
+            lst_blocks=[
+                self.format_obj(
+                    key=f"Getting the IP of {self._interface}:",
+                    value_type=HardwareTestJsonParamType.STRING,
+                    value=response,
+                ),
+            ],
+        )
+
+
+class HardwareTestBattery(HardwareTest):
+    def __init__(self) -> None:
+        super().__init__(service_identifier="tests/battery")
+
+    def test_id(self) -> str:
+        return "Battery"
+
+    def test_description_preparation(self) -> str:
+        return self.html_util_ul(["Place your Duckiebot near the charging cable."])
+
+    def test_description_expectation(self) -> str:
+        return self.html_util_ul(
+            [
+                "The battery firmware version should be at least <code>2.0.2</code>.",
+                "The PCB version should be at least <code>16</code>.",
+                "The test result should reflect the accurate charging status, within 2 seconds of a change.",
+                "Only mark the Success when the version examinations pass, and the charging states are indicated correctly.",
+            ]
```

Comment:
"The battery firmware version should be at least <code>2.0.2</code>.",
"The PCB version should be at least <code>16</code>.",
Run the test first with the charging cable plugged in.  Then unplug the charging cable, wait 5 seconds and rerun the test.  If the versions and charging state are correct for both tests, you can then mark the test as Success.